app/src/main/java/unipd/se18/ocrcamera/GalleryActivity.java

```diff
@@ -0,0 +1,385 @@
+                builder = new AlertDialog.Builder(getContext(), android.R.style.Theme_Material_Dialog_Alert);
+            } else {
+                builder = new AlertDialog.Builder(getContext());
+            }
+            builder.setTitle("Error")
+                    .setMessage("Error while deleting the photo. Retry")
+                    .setPositiveButton(android.R.string.yes, new DialogInterface.OnClickListener() {
+                        public void onClick(DialogInterface dialog, int which) {
+                            //I'm not performing any action because this allert is just for provide information
+                        }
+                    })
+                    .setIcon(android.R.drawable.ic_dialog_alert)
+                    .show();
+        }
+    }
+
+
+    /*********************************************/
+    /******************PERMISSIONS****************/
+    /*********************************************/
+
+
+    /**
+     * Verify if the user granted the permission
+     */
+    @Override
+    public void onRequestPermissionsResult(int requestCode, String[] permissions, int[] grantResults)
+    {
+        if(requestCode==REQUEST_PERMISSION_CODE)
+        {
+            // If request is cancelled, the result arrays are empty.
+            if (grantResults.length > 0 && grantResults[0] != PackageManager.PERMISSION_GRANTED) {
+                // permission denied
+                showPermissionErrorDialog();
+            }
+            else
+            {
+                loadHomeFragment();
+            }
+        }
+    }
+
+    /**
+     * Verify if the app has the storage permission
+     * @return boolean of the current status (before asking the permission)
+     * if its false this prevent the app to load the fragment that uses the internal storage
+     */
+    public boolean verifyStoragePermission()
+    {
+        //Check and in case Ask for permission
+        if (ContextCompat.checkSelfPermission(this, Manifest.permission.WRITE_EXTERNAL_STORAGE) != PackageManager.PERMISSION_GRANTED) {
+            ActivityCompat.requestPermissions(this,
+                    new String[]{Manifest.permission.WRITE_EXTERNAL_STORAGE}, REQUEST_PERMISSION_CODE);
+            return false;
+        }
+        else
+        {
+            return true;
+        }
+    }
+
+    /**
+     * Show a simple error message before closing the activity in case bad permission
+     */
+    private void showPermissionErrorDialog()
+    {
+        AlertDialog.Builder builder;
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            builder = new AlertDialog.Builder(this, android.R.style.Theme_Material_Dialog_Alert);
+        } else {
+            builder = new AlertDialog.Builder(this);
+        }
+        builder.setTitle("Permission error")
+                .setMessage("You did not have authorized the app to use internal storage. The app will be closed.")
```

Comment:
In order to avoid hard coding and to favourite the translations, could a good idea be putting these strings to the strings.xml file?